File: partitioning-into-minimum-number-of-deci-binary-numbers/partitioning-into-minimum-number-of-deci-binary-numbers.cpp
class Solution {
public:
    int minPartitions(string n) {
        int ans = -1;
        int len = n.size();
        for(int i = 0 ; i < len ; i++){
            ans = max(ans,n[i] - '0');
        }
        return ans;
    }
};

File: partitioning-into-minimum-number-of-deci-binary-numbers/partitioning-into-minimum-number-of-deci-binary-numbers.go
func max(i,j int)int{
    if i < j {
        return j
    }
    return i
}
func minPartitions(n string) int {
    ans := -1
    length := len(n)
    for i := 0 ; i < length ; i++ {
        ans = max(ans,int(n[i]) - int('0'))
    } 
    return ans
}